src/libcollectionstest/vec.rs

```diff
@@ -439,6 +439,16 @@ fn test_drain_range() {
 }
 
 #[test]
+fn splice() {
+    let mut v = vec![1, 2, 3, 4, 5];
+    let a = [10, 11, 12];
+    v.splice(2..4, a.iter().cloned());
+    assert_eq!(v, &[1, 2, 10, 11, 12, 5]);
+    v.splice(1..3, Some(20));
+    assert_eq!(v, &[1, 20, 11, 12, 5]);
+}
```

Comment:
These are some pretty small tests for a pretty meaty `unsafe` implementation, can you add some more "flavorful" tests along the lines of:

* Returning some surprising `size_hint` values from the "to replace" iterator
* Forgetting the splice iterator
* Panicking during iteration of the splice iterator
* Panicking while draining elements

etc